### PR TITLE
Added support for Python 3

### PR DESCRIPTION
**Pull request by namrata4**
- used six for ensuring python 2 and 3 compatibility
- changes made for StringIO

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,3 +5,4 @@ mock==1.3.0
 nose==1.3.4
 python_swiftclient==2.7.0
 rednose==1.1.1
+six
```
Comment (b4hand) [inline requirements.txt]: Please add a concrete version to `requirements.txt` Also, `six` will need to be added to the `setup.py` as well.

**Comment by b4hand** [comment]
It looks like there are a few more changes that need to be made for Python 3 support. The code is currently using `xrange`. `from six.moves import range` and then replace all instances of `xrange` with `range`.

**Comment by namrata4** [comment]
Looks like it isn't issue of only StringIO (I was hoping for a quick fix), some other things also needs to be changed like basestring and xrange. Making changes for that.

**Comment by neilmb** [comment]
@b4hand You had requested some changes. I would love to have a Python 3 compatible version of s3po released. Would you re-review the current state of this and once merged we could push a PyPI release? Perhaps we want to update some classifiers in `setup.py` or tout the Python 3 compatibility in `README.md` as well?

**Comment by b4hand** [comment]
Sorry, it looks like this just dropped out of my visibility and got forgotten. @neilmb: Feel free to update the version and classifiers in `setup.py`, and I can get a version pushed out.

**Pull request merged**